diff -rupN snapper-0.2.5.old/pam/Makefile.am snapper-0.2.5.new/pam/Makefile.am
--- snapper-0.2.5.old/pam/Makefile.am	2015-01-19 16:55:45.000000000 +0100
+++ snapper-0.2.5.new/pam/Makefile.am	2015-01-30 16:31:43.122790381 +0100
@@ -8,7 +8,7 @@ AM_CFLAGS = -D_GNU_SOURCE -Wwrite-string
 
 AM_CPPFLAGS = -I$(top_srcdir) $(DBUS_CFLAGS)
 
-securelibdir = $(shell echo /`basename $(libdir)`/security)
+securelibdir = $(libdir)/security
 
 securelib_LTLIBRARIES = pam_snapper.la
 
