diff -rupN snapper-0.1.7.old/configure.ac snapper-0.1.7/configure.ac
--- snapper-0.1.7.old/configure.ac	2014-10-08 14:39:09.033536144 +0200
+++ snapper-0.1.7/configure.ac	2014-10-08 14:47:11.087815498 +0200
@@ -103,6 +103,7 @@ AC_ARG_ENABLE([zypp], AC_HELP_STRING([--
 AM_CONDITIONAL(HAVE_ZYPP, [test "x$with_zypp" = "xyes"])
 
 AC_CHECK_LIB(btrfs, btrfs_read_and_process_send_stream)
+AC_CHECK_HEADERS([btrfs/version.h])
 
 AC_ARG_ENABLE([xattrs], AC_HELP_STRING([--disable-xattrs],[Disable extended attributes support]),
 		[with_xattrs=$enableval],[with_xattrs=yes])
diff -rupN snapper-0.1.7.old/snapper/Btrfs.cc snapper-0.1.7/snapper/Btrfs.cc
--- snapper-0.1.7.old/snapper/Btrfs.cc	2014-10-08 14:39:09.027536166 +0200
+++ snapper-0.1.7/snapper/Btrfs.cc	2014-10-08 14:48:21.126565503 +0200
@@ -31,6 +31,11 @@
 #include <sys/ioctl.h>
 #include <asm/types.h>
 #ifdef HAVE_LIBBTRFS
+#ifdef HAVE_BTRFS_VERSION_H
+#include <btrfs/version.h>
+#else
+#define BTRFS_LIB_VERSION (100)
+#endif
 #include <btrfs/ioctl.h>
 #include <btrfs/send.h>
 #include <btrfs/send-stream.h>
@@ -1002,7 +1007,13 @@ namespace snapper
 	    boost::this_thread::interruption_point();
 
 	     // remove the fourth parameter for older versions of libbtrfs
-	    int r = btrfs_read_and_process_send_stream(fd, &send_ops, &*this, 0);
+	    int r;
+
+#if BTRFS_LIB_VERSION < 101
+	    r = btrfs_read_and_process_send_stream(fd, &send_ops, &*this, 0);
+#else
+	    r = btrfs_read_and_process_send_stream(fd, &send_ops, &*this, 0, 1);
+#endif
 
 	    if (r < 0)
 	    {
