--- a/doc/snapper.xml.in	2013-07-12 10:47:19.000000000 +0200
+++ b/doc/snapper.xml.in	2013-07-29 12:43:31.582526830 +0200
@@ -39,9 +39,7 @@
 
     <para>Snapper never modifies the content of snapshots. Thus snapper creates
     read-only snapshots if supported by the kernel. Supported filesystems are
-    btrfs and ext4 as well as snapshots of LVM logical volumes with
-    thin-provisioning. Some filesystems might not be supported depending on your
-    installation.</para>
+    btrfs as well as snapshots of LVM logical volumes with thin-provisioning.</para>
   </refsect1>
 
   <refsect1 id='concepts'>
@@ -235,7 +233,7 @@
 	    <varlistentry>
 	      <term><option>-f, --fstype</option> <replaceable>fstype</replaceable></term>
 	      <listitem>
-		<para>Manually set filesystem type. Supported values are btrfs, ext4 and lvm. For
+		<para>Manually set filesystem type. Supported values are btrfs and lvm. For
 		lvm, snapper uses LVM thin-provisioned snapshots. The filesystem type on top of
 		LVM must be provided in parentheses, e.g. lvm(xfs).</para>
 		<para>Without this option snapper tries to detect the filesystem.</para>
